### PR TITLE
Leaderboard integration closes #130

### LINKED ISSUE #130: Make a leaderboard (backend)

### PR DESCRIPTION
**Pull request by GijsWeterings**

### CODE REVIEW

**Comment by JWGmeligMeyling** [comment]
Looks good!

File: src/test/java/me/moodcat/database/controllers/UserDAOTest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        final int expectedNumberOfUsers = 2;
+
+        assertEquals(expectedNumberOfUsers, userDAO.getLeaderboard(filter).size());
+    }
+
+    @Test
+    @TestBootstrap("/bootstrap/users.json")
+    public void testOrder() {
+        final int filter = 2;
+        ArrayList<User> bestUsers = (ArrayList<User>) userDAO.getLeaderboard(filter);
```
Comment (JWGmeligMeyling) [inline src/test/java/me/moodcat/database/controllers/UserDAOTest.java]: A reqular list wouldn't require a cas. Or is this test dependent on arraylists?

**Comment by Azkle** [comment]
It all works as expected and the code seems fine.

**Comment by Jaapp-** [comment]
Looks solid.

**Comment by JeremybellEU** [comment]
This PR should be rebased to master in order to be mergable.

**Comment by JWGmeligMeyling** [comment]
Rebased in https://github.com/MoodCat/MoodCat.me-Core/pull/145

**Pull request closed without merging**